### PR TITLE
Add four alibabacloud-aisearch llama-index integrations: rerank, node_parser, readers, embeddings

### PR DESCRIPTION
**Pull request by maocaisheng**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by maocaisheng** [comment]
To enhance efficiency, we have integrated four features in this merge. Please take some time to review them. Thank you!

File: llama-index-integrations/embeddings/llama-index-embeddings-alibabacloud-aisearch/llama_index/embeddings/alibabacloud_aisearch/base.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        GetTextEmbeddingRequest,
+        GetTextEmbeddingResponse,
+    )
+    from alibabacloud_tea_openapi.models import Config as AISearchConfig
+    from alibabacloud_searchplat20240529.client import Client
+    from Tea.exceptions import TeaException
+except ImportError:
+    raise ImportError(
+        "Could not import alibabacloud_searchplat20240529 python package. "
+        "Please install it with `pip install alibabacloud_searchplat20240529`."
```
Comment (logan-markewich) [inline llama-index-integrations/embeddings/llama-index-embeddings-alibabacloud-aisearch/llama_index/embeddings/alibabacloud_aisearch/base.py:24]: Is that actually the name of the package? Thats wild haha

Comment (maocaisheng) [reply llama-index-integrations/embeddings/llama-index-embeddings-alibabacloud-aisearch/llama_index/embeddings/alibabacloud_aisearch/base.py:24]: Yes, it is. haha

**Review by logan-markewich — APPROVED** [review]
Honestly this looks pretty good to me -- great work!

**Pull request merged**